### PR TITLE
Adding linting to CI and Makefile

### LINKED ISSUE #49: Missing CI
CI should run `gopnm` so stuff like:
```
github.com/jbuchbinder/gopnm is not in your go.mod file.
```
doesn't slip through.

It should probably also run `govet`

### PR DESCRIPTION
**Pull request by local-minimum**
Resolves #49 

If there are more things that should be run, we can add them later.

I don't know why but `make lint` doesn't work on the github action.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,15 +1,21 @@
 # Go parameters
 GOCMD=go
+GOPATH ?= ~/go/bin
```
Comment (joakimmoller) [inline Makefile]: You might confuse this with PATH? setting GOPATH like this will break my environment.
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
 GOBUILD=$(GOCMD) build
 GOCLEAN=$(GOCMD) clean
 GOTEST=$(GOCMD) test
+GOVET=${GOCMD} vet
+GOLINT=$(GOPATH)/golint
```
Comment (joakimmoller) [inline Makefile]:
```suggestion
GOLINT=golint
```

File: go.mod [hunk 3]
```diff
@@ -5,4 +5,5 @@ go 1.14
 require (
 	github.com/howeyc/crc16 v0.0.0-20171223171357-2b2a61e366a6
 	github.com/jbuchbinder/gopnm v0.0.0-20150223212718-5176c556b9ce
+	golang.org/x/lint v0.0.0-20200302205851-738671d3881b // indirect
```
Comment (joakimmoller) [inline go.mod:8]: golint is a tool? that shouldn't be in the app?
```suggestion
```

File: go.sum [hunk 4]
```diff
@@ -4,3 +4,17 @@ github.com/jbuchbinder/gopnm v0.0.0-20150223212718-5176c556b9ce h1:SHuC5z27u3lD7
...
+golang.org/x/mod v0.1.1-0.20191105210325-c90efee705ee/go.mod h1:QqPTAvyqsEbceGzBzNggFXnrqF1CaUcvgkdR5Ot7KZg=
+golang.org/x/net v0.0.0-20190404232315-eb5bcb51f2a3/go.mod h1:t9HGtf8HONx5eT2rtn7q6eTqICYqUVnKs3thJo3Qplg=
+golang.org/x/net v0.0.0-20190620200207-3b0461eec859/go.mod h1:z5CRVTTTmAJ677TzLLGU+0bjPO0LkuOLi4/5GtJWs/s=
+golang.org/x/sync v0.0.0-20190423024810-112230192c58/go.mod h1:RxMgew5VJxzue5/jJTE5uejpjVlOe/izrB70Jof72aM=
+golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a/go.mod h1:STP8DvDyc/dI5b8T5hshtkjS+E42TnysNCUPdjciGhY=
+golang.org/x/sys v0.0.0-20190412213103-97732733099d/go.mod h1:h1NjWce9XRLGQEsW7wpKNCjG9DtNlClVuFLEZdDNbEs=
+golang.org/x/text v0.3.0/go.mod h1:NqM8EUOU14njkJ3fqMW+pc6Ldnwhi/IjpwHt7yyuwOQ=
+golang.org/x/tools v0.0.0-20200130002326-2f3ba24bd6e7 h1:EBZoQjiKKPaLbPrbpssUfuHtwM6KV/vb4U85g/cigFY=
+golang.org/x/tools v0.0.0-20200130002326-2f3ba24bd6e7/go.mod h1:TB2adYChydJhpapKDTa4BR/hXlZSLoq2Wpct/0txZ28=
+golang.org/x/xerrors v0.0.0-20191011141410-1b5146add898/go.mod h1:I/5z698sn9Ka8TeJc9MKroUUfqBBauWjQqLJ2OPfmY0=
```
Comment (joakimmoller) [inline go.sum:20]:
```suggestion
```

**Review by joakimmoller — CHANGES_REQUESTED** [review]
clean it up a bit, maybe it is easier to use a github action to lint on github?

File: Makefile (hunk 1 above)
Comment (local-minimum) [reply Makefile]: No, not really and isn't this just setting it if it isn't already set? Maybe though it should have been `GOPATH ?= ~/go` instead?

File: go.mod (hunk 3 above)
Comment (local-minimum) [reply go.mod:8]: I figure that's the `// indirect` bit that it is a dev-dependency, these were added automatically by the tools in VSC.

File: Makefile (hunk 1 above)
Comment (local-minimum) [reply Makefile]: No PATH would setting would be useless here, `go` is already in the path naturally or nothing in the make script would have worked.

**Comment by joakimmoller** [comment]
Closing this...

**Pull request closed without merging**